### ISSUE TITLE
Ghost workspaceitems after rejected in workflow and submitter deleted

### ISSUE THREAD

**Issue by MarieVerdonck** [DSpace/DSpace#8641]
**Describe the bug**
If a submitter user is deleted/removed when it still has submitted items in the workflow and these items get rejected during the workflow they get sent back to the workspace, as workspace items. This results in 'ghost' workspace items that will never be removed/submitted. They can still be visited and edited by an admin, but they won't render on its MyDSpace page so they won't be aware they exist, nor can an admin actually submit them again (only read no edit access likely).

**To Reproduce**
1. (As admin) Create a new submitter user
2. As that new submitter submit a new item in a collection with workflow (collection that has workflow groups in its edit form > assign roles tab defined, with at least 1 member) 
3. (As reviewer) verify this item is now in workflow and can be claimed and then accepted or rejected by reviewer
4. (As admin) Delete the original submitter user
5. (As reviewer) Reject the previously submitted item that is now in workflow with a reason
6. Verify there now exists a 'ghost' workspace item (workspace item that can't be discovered, submitted or deleted via UI) for this test item

These 'ghost' workspace items are still accessible by admin in e.g.:

- `{ui.url}/workspaceitems/{workspaceitem_id}/view`
- `{ui.url}/workspaceitems/{workspaceitem_id}/edit`
- `{rest.url}/server/api/submission/workspaceitems/{workspaceitem_id}`

**Expected behavior**
Workspace items whose submitter no longer exists should not exist either.

**Comment by tdonohue** [comment]
Agreed, this sounds like a bug.  If you delete a Submitter, it likely should also delete any WorkspaceItems or WorkflowItems associated with that submitter (We may need to add a warning to that effect in the UI during deletion of a user which has Workspace/Workflow items.)

**Comment by buithaihai** [comment]
DLCorp would like to claim this issue, our estimate is 12 hours of coding & testing.
